docs/references/react-router/root-auth-loader.mdx

```diff
@@ -0,0 +1,159 @@
+
+export async function loader(args: Route.LoaderArgs) {
+  return rootAuthLoader(args, { signInUrl: '/sign-in' })
+}
+```
+
+### Loading additional data
+
+If you need to load additional data, you can pass your loader directly to `rootAuthLoader()` as the second argument. The options object is passed as the third argument.
+
+```tsx {{ filename: 'app/root.tsx', mark: [7, 13] }}
+import { rootAuthLoader } from '@clerk/react-router/ssr.server'
+import type { Route } from './+types/root'
+
+export async function loader(args: Route.LoaderArgs) {
+  return rootAuthLoader(
+    args,
+    ({ request, context, params }) => {
+      // Loader
+      const { userId } = request.auth
+
+      return { userId }
+    },
+    { signInUrl: '/sign-in' }, // Options
+  )
+}
+```
+
+## `rootAuthLoader()` options
+
+<Properties>
+  - `secretKey?`
+  - `string`
+
+  The Clerk Secret Key from the [**API keys**](https://dashboard.clerk.com/last-active?path=api-keys) page in the Clerk Dashboard.
+
+  ---
+
+  - `jwtKey?`
+  - `string`
+
+  The PEM public key from the **[**API keys**](https://dashboard.clerk.com/last-active?path=api-keys) page -> Show JWT public key -> PEM Public Key** section in the Clerk Dashboard. For more information, refer to [Manual JWT verification](/docs/backend-requests/handling/manual-jwt).
+
+  ---
+
+  - `publishableKey?`
+  - `string`
+
+  The Clerk Publishable Key from the [**API keys**](https://dashboard.clerk.com/last-active?path=api-keys) page in the Clerk Dashboard.
+
+  ---
+
+  - `domain?`
+  - `string`
+
+  The domain of a [satellite application](/docs/advanced-usage/satellite-domains) in a multi-domain setup.
+
+  ---
+
+  - `isSatellite?`
+  - `boolean`
+
+  Whether the instance is a satellite domain in a multi-domain setup. Defaults to `false`.
+
+  ---
+
+  - `proxyUrl?`
+  - `string`
+
+  The proxy URL from a multi-domain setup.
+
+  ---
+
+  - `sdkMetadata?`
+  - `{ name: string, version: string }`
+
+  Metadata about the SDK.
+
+  ---
+
+  - `telemetry?`
+  - `{ disabled: boolean, debug: boolean }`
+
+  [Telemetry](/docs/telemetry) configuration.
+
+  ---
+
+  - `userAgent?`
+  - `string`
+
+  The User-Agent request header passed to the Clerk API.
+
+  ---
+
+  - `apiUrl?`
+  - `string`
+
+  The [Clerk Backend API](/docs/reference/backend-api){{ target: '_blank' }} endpoint. Defaults to `'https://api.clerk.com'`.
+
+  ---
+
+  - `apiVersion?`
+  - `string`
+
+  The version passed to the Clerk API. Defaults to `'v1'`.
+
+  ---
+
+  - `audience?`
+  - `string | string[]`
+
+  A string or list of [audiences](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3).
+
+  ---
+
+  - `authorizedParties?`
+  - `string[]`
+
+  An allowlist of origins to verify against, to protect your application from the subdomain cookie leaking attack.<br />For example: `['http://localhost:3000', 'https://example.com']`
+
+  ---
+
+  - `signInUrl?`
+  - `string`
+
+  An alternative sign in URL.
+
+  ---
+
+  - `signUpUrl?`
+  - `string`
+
```

Comment:
what does passing `signInUrl` or `signUpUrl` to the `rootAuthLoader()` do? does it override account portal, and set the app's sign-up and sign-in URLs?

Reply:
Discussed in private messages:
Lennart: "The signInUrl option is the same as the CLERK_SIGN_IN_URL env var, so typically people won't set them"

discussed to use "The full URL or path to the sign in page. Use this property to provide the target of the 'Sign In' link that's rendered. It's recommended to use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead." as the prop description instead

[fd668b0](https://github.com/clerk/clerk-docs/pull/1760/commits/fd668b00d92c1214f97a75484ab4f8c689e26581)